script.js

```diff
@@ -1,82 +1,217 @@
+      let episodeSectionElement = document.getElementById("displayEpisode");
+      removeChild(episodeSectionElement);
+      document.getElementById("showSelect").value = element.id;
+      fetchEpisodes(url);
+    });
   });
 }
 
-function makeSearchSection() {
-  let searchSection = document.getElementById("searchSection");
+// Make section for search episode
+function makeSearchEpisodeSection() {
+  const rootElem = document.getElementById("root");
+  let searchSection = document.createElement("section");
+  searchSection.id = "searchSection";
+  searchSection.className = " px-2 mx-auto py-4";
+  searchSection.style.display = "none";
+  rootElem.append(searchSection);
+  let parentOfSearchElements = document.createElement("div");
+  parentOfSearchElements.className = "row g-1 mx-auto";
+  parentOfSearchElements.id = "parentOfSearchElements";
+  searchSection.append(parentOfSearchElements);
+  let parentOfShowSelect = document.createElement("div");
+  parentOfShowSelect.className = "col-12 col-sm-6 col-md-3";
+  let showSelectElement = document.createElement("select");
+  showSelectElement.id = "showSelect";
+  showSelectElement.className = "form-control m-1 p-2";
+  parentOfShowSelect.append(showSelectElement);
+  let parentOfEpisodeSelect = document.createElement("div");
+  parentOfEpisodeSelect.className = "col-12 col-sm-6 col-md-3";
+  let episodeSelectElement = document.createElement("select");
+  episodeSelectElement.className = " form-control m-1 p-2 ";
+  episodeSelectElement.id = "selectEpisode";
+  parentOfEpisodeSelect.append(episodeSelectElement);
+  let parentOfInputElement = document.createElement("div");
+  parentOfInputElement.className = "col-md-2 col-12 col-sm-6 ";
   let inputElement = document.createElement("input");
   inputElement.id = "searchKey";
-  inputElement.className = "col-3 m-3  form-control";
+  inputElement.className = "form-control m-1 ";
+  parentOfInputElement.append(inputElement);
   let labelElement = document.createElement("label");
   labelElement.id = "searchComment";
-  labelElement.className = "col-4 m-3 text-center";
-  searchSection.append(inputElement, labelElement);
-  inputElement.addEventListener("input", search);
+  labelElement.className = "col-md-2 col-12 col-sm-5 m-1 p-1 pl-3 text-left";
+  let backBtnElement = document.createElement("button");
+  backBtnElement.id = "btn";
+  backBtnElement.className = "col-md-1 col-11 mt-3 m-md-1 mx-auto btn";
+  backBtnElement.innerHTML = "Back";
+  backBtnElement.addEventListener("click", displayShowsSection);
+  parentOfSearchElements.append(
+    parentOfShowSelect,
+    parentOfEpisodeSelect,
+    parentOfInputElement,
+    labelElement,
+    backBtnElement
+  );
+  inputElement.addEventListener("input", searchEpisode);
 }
 
-function search() {
+// Search episode
+function searchEpisode() {
   let episodeFilter = allEpisodes.filter(filterByKey);
-  const rootElem = document.getElementById("root");
```

Comment:
I have noticed you used some of the variables like rootElem a few times in your code. I think it would be better to put them in the global scope to have a concise code.

Reply:
Thank you Daniel,
You are right but in the last session we were recommended to not using global element so I removed most of them.

Reply:
I am not 100% sure but the thing is if you are using a variable many times it's better to put it in the global scope to have a DRY code.